tests/apollo/test_skvbc_control_commands.py

```diff
@@ -0,0 +1,151 @@
+# You may not use this product except in compliance with the Apache 2.0 License.
+#
+# This product may include a number of subcomponents with separate copyright
+# notices and license terms. Your use of these subcomponents is subject to the
+# terms and conditions of the subcomponent's license, as noted in the LICENSE
+# file.
+import os.path
+import random
+import unittest
+from os import environ
+
+import trio
+
+from util import blinking_replica
+from util import skvbc as kvbc
+from util.bft import with_trio, with_bft_network, KEY_FILE_PREFIX
+
+
+def start_replica_cmd(builddir, replica_id):
+    """
+    Return a command that starts an skvbc replica when passed to
+    subprocess.Popen.
+
+    Note each arguments is an element in a list.
+    """
+    statusTimerMilli = "500"
+    viewChangeTimeoutMilli = "10000"
+    path = os.path.join(builddir, "tests", "simpleKVBC", "TesterReplica", "skvbc_replica")
+    return [path,
+            "-k", KEY_FILE_PREFIX,
+            "-i", str(replica_id),
+            "-s", statusTimerMilli,
+            "-v", viewChangeTimeoutMilli,
+            "-p" if os.environ.get('BUILD_ROCKSDB_STORAGE', "").lower() in set(["true", "on"]) else "",
+            "-t", os.environ.get('STORAGE_TYPE')]
+
+
+class SkvbcControlCommandsTest(unittest.TestCase):
+    @with_trio
+    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: n == 7)
+    async def test_wedge_command(self, bft_network):
+        """
+             Sends a wedge command and check that the system stops from processing new requests.
+             Note that in this test we assume no failures and synchronized network.
+             The test does the following:
+             1. A client sends a wedge command
+             2. The client verify that the system reached to a super stable checkpoint
+             3. The client tries to initiate a new write bft command and fails
+         """
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+        client = bft_network.random_client()
+
+        checkpoint_before = await bft_network.wait_for_checkpoint(replica_id=0)
+
+        await client.write(skvbc.write_req([], [], block_id=0, wedge_command=True))
+
+        for replica_id in range(bft_network.config.n):
+            with trio.fail_after(seconds=30):
+                while True:
+                    with trio.move_on_after(seconds=1):
+                        checkpoint_after = await bft_network.wait_for_checkpoint(replica_id=replica_id)
+                        if checkpoint_after == checkpoint_before + 2:
+                            break
+
+        await self.validate_stop_on_super_stable_checkpoint(bft_network, skvbc)
+
+    # @unittest.skip("Test case assumes the happy path")
+    @with_trio
+    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: n == 7)
+    async def test_wedge_command_with_state_transfer(self, bft_network):
```

Comment:
It could be interesting to run this test as a separate test suite in Hermes (similar to [ApolloBftTests](https://gitlab.eng.vmware.com/blockchain/vmwathena_blockchain/blob/master/hermes/suites/apollo_bft_tests.py#L78)). Those tests run SKVBC in Concord, so they do use TLS instead of UDP. I expect this would help bring clarity on the missing messages problem (if still applicable) ...

Reply:
Yeah, that is a great idea!
Let's take it offline